GoogleSignIn/Tests/Unit/GIDAuthorizationResponseHelperTest.m

```diff
@@ -0,0 +1,169 @@
+                                                               configuration:nil
+                                                                       error:nil];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.code, expectedError.code);
+  XCTAssertEqualObjects(authFlow.error.userInfo[NSLocalizedDescriptionKey],
+                        expectedError.userInfo[NSLocalizedDescriptionKey]);
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+}
+
+- (void)testGenerateAuthFlowWithMissingAuthorizationResponse {
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+  NSError *error = [NSError errorWithDomain:kGIDVerifyErrorDomain
+                                       code:GIDVerifyErrorCodeUnknown
+                                   userInfo:nil];
+  GIDAuthorizationResponseHandler *responseHandler =
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:nil
+                                                                    flowName:GIDFlowNameVerify
+                                                               configuration:nil
+                                                                       error:error];
+
+  GIDAuthFlow *authFlow = [responseHandler generateAuthFlowFromAuthorizationResponse];
+  XCTAssertNotNil(authFlow);
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertEqual(authFlow.error.code, error.code);
+#endif // TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+}
+
+- (void)testMaybeFetchTokenWithAuthFlowError {
+  NSError *error = [NSError errorWithDomain:kGIDSignInErrorDomain
+                                       code:kGIDSignInErrorCodeUnknown
+                                   userInfo:nil];
+  GIDAuthFlow *authFlow = [[GIDAuthFlow alloc] initWithAuthState:nil
+                                                           error:error
+                                                      emmSupport:nil
+                                                     profileData:nil];
+  GIDAuthorizationResponseHandler *responseHandler = 
+      [[GIDAuthorizationResponseHandler alloc] initWithAuthorizationResponse:nil
+                                                                  emmSupport:kEMMVersion
+                                                                    flowName:GIDFlowNameSignIn
+                                                               configuration:nil
+                                                                       error:nil];
+
+  [responseHandler maybeFetchToken:authFlow];
+  XCTAssertNil(authFlow.authState);
+  XCTAssertNotNil(authFlow.error);
+  XCTAssertNil(authFlow.emmSupport);
+}
+
+- (void)testMaybeFetchTokenWithExpirationError {
```

Comment:
I don't understand the name of this test method and what's being done inside of it. I expect some kind of assert around an expiration error, but I don't see that.

Reply:
Changed to `testMaybeFetchToken_noRefresh` indicating that this will test `-[GIDAuthorizationResponseHandler maybeFetchToken:]` when the access token isn't near expiration. Let me know if you have a suggestion.

Reply:
LGTM